pallets/omnipool-liquidity-mining/src/lib.rs

```diff
@@ -1125,4 +1205,64 @@ impl<T: Config> Pallet<T> {
 
 		Ok((deposit_id, lp_position))
 	}
+
+	#[require_transactional]
+	fn do_withdraw_shares(
+		origin: OriginFor<T>,
+		deposit_id: DepositId,
+		yield_farm_id: YieldFarmId,
+	) -> Result<Balance, DispatchError> {
+		let owner = Self::ensure_nft_owner(origin, deposit_id)?;
+
+		//NOTE: not tested - this should never fail.
```

Comment:
What do you mean by this comment? If user provides wrong deposit_id you'll get error. But it looks like it's handled correctly.

Reply:
These comments were added by @martinfridrich in several places in the past when the pallet was made. I think it is for explaining the reason for using defensive_ok and returning `InconsistentStateError`.

Reply:
it's just comment why this error is not teste. It doesn't have to be there as `InconsistentStateError` represents error that can never happen. 
It's probably relic from before I added `InconsistentStateError` errors.

@jak-pan this storage read can't fail. If user provide wrong `deposit_id` it would fail on `ensure_nft_owner`
Fail of this storage read would mean that LM's position exists without underlying omnipool's position and it should never happen. 

@dmoka why did you add this function? I quickly grep-ed it and it looks like it's not used by this pallet and it's not `pub`